tls.go

```diff
@@ -21,27 +23,171 @@ func tlsDirective(m *config.Map, node *config.Node) (interface{}, error) {
 		case "off":
 			return nil, nil
 		case "self_signed":
-			cfg := &tls.Config{}
+			cfg := &tls.Config{
+				MinVersion: tls.VersionTLS10,
+				MaxVersion: tls.VersionTLS12,
+			}
 			if err := makeSelfSignedCert(cfg); err != nil {
 				return nil, err
 			}
+			log.Debugln("tls: using self-signed certificate, this is not secure!")
 			return &cfg, nil
 		}
 	case 2:
-		cfg := tls.Config{}
-		if cert, err := tls.LoadX509KeyPair(node.Args[0], node.Args[1]); err != nil {
+		cfg, err := readTLSBlock(node)
+		if err != nil {
 			return nil, err
-		} else {
-			cfg.Certificates = append(cfg.Certificates, cert)
-			return &cfg, nil
 		}
+		return cfg, nil
 	default:
 		return nil, m.MatchErr("expected 1 or 2 arguments")
 	}
 
 	return nil, nil
 }
 
+var strVersionsMap = map[string]uint16{
+	"tls1.0": tls.VersionTLS10,
+	"tls1.1": tls.VersionTLS11,
+	"tls1.2": tls.VersionTLS12,
+	"tls1.3": tls.VersionTLS13,
+	"":       0, // use crypto/tls defaults if value is not specified
+}
+
+var strCiphersMap = map[string]uint16{
+	// TLS 1.0 - 1.2 cipher suites.
+	"RSA-WITH-RC4128-SHA":                0x0005,
```

Comment:
There are also constants for these (`tls.TLS_RSA_WITH_RC4_128_SHA`)

Reply:
(I prefer to use constants because it would be unfortunate to map a cipher to a weaker one by mistake)